### PR TITLE
Integrate storage with commands

### LINKED ISSUE #28: Implement exit module command

### PR DESCRIPTION
**Pull request by 8kdesign**
Fixes #28 , #37, #38, #39

### CODE REVIEW

File: src/main/java/seedu/duke/module/ModuleList.java [hunk 1]
```diff
@@ -40,17 +40,15 @@ public static void loadModuleNames(Loader loader) {
...
+            Writer writer = new Writer();
+            writer.createFile(moduleCode);
+            return true;
         }
-    }
-    
-    public void addModule(String moduleCode) {
-        modules.add(moduleCode);
+        return false;
     }
```
Comment (aliciatay-zls) [inline src/main/java/seedu/duke/module/ModuleList.java:52]: Thank you for removing these duplicated methods!!

File: src/test/java/seedu/duke/module/ModuleListTest.java [hunk 2]
```diff
@@ -131,103 +131,88 @@ void setSelectedModule_invalidContent_() throws IOException {
...
-        File directory = new File(FOLDER_PATH);
-        File[] files = directory.listFiles();
-        if (files == null) {
-            return;
-        }
-        for (File file : files) {
-            file.delete();
-        }
-        directory.delete();
-    }
```
Comment (aliciatay-zls) [inline src/test/java/seedu/duke/module/ModuleListTest.java:232]: Can add a newline at the end for this file

File: src/test/java/seedu/duke/commands/CommandTestUtil.java [hunk 3]
```diff
@@ -13,11 +14,11 @@
...
-    public static ModuleList initialiseModuleList() {
-        ModuleList modules = new ModuleList();
-        Module module = new Module("CS2113T");
-        modules.selectedModule = module;
-        return modules;
+    public static void initialiseModuleList() {
+        TestUtil.removeFiles();
+        ModuleList.loadModuleNames();
+        ModuleList.addModule("CS2113T");
+        ModuleList.setSelectedModule("CS2113T");
```
Comment (aliciatay-zls) [inline src/test/java/seedu/duke/commands/CommandTestUtil.java:21]: If this is not used anywhere can just delete

**Review by aliciatay-zls — COMMENTED** [review]
LGTM, cleaned up code looks very good to me too!

**Review by aliciatay-zls — APPROVED** [review]

File: src/main/java/seedu/duke/Duke.java [hunk 4]
```diff
@@ -35,23 +30,38 @@ private static void start() {
...
+
+        Parser parser = new Parser();
+
         //Loop
         while (!isExit) {
+            if (ModuleList.getSelectedModule() == null) {
+                System.out.print("<GULIO>");
+            } else {
+                System.out.print("<" + ModuleList.getSelectedModule().getModuleCode() + ">");
+            }
```
[the remark this answers is not in the document]
Comment (8kdesign) [reply src/main/java/seedu/duke/Duke.java:48]: Sure

File: src/test/java/seedu/duke/commands/CommandTestUtil.java (hunk 3 above)
Comment (8kdesign) [reply src/test/java/seedu/duke/commands/CommandTestUtil.java:21]: Ok removed.

**Pull request merged**